### ISSUE TITLE
Keka main window always opens, prevents automatic quit

### ISSUE THREAD

**Issue by kristofferR** [aonez/Keka#38]
I've set Keka to close automatically when no window is open and that works fine. However, since Keka opens the compression window at every launch (even when Keka opens in order to decompress an archive), Keka stays open after extraction is finished.

This is quite annoying, and makes me use another app as my default decompression app instead of Keka. Keka should quit automatically after extraction when it has been launched to extract something, that should at least be an option.

**Comment by aonez** [comment]
That is in fact the current behaviour. If you double click a compressed file, Keka will launch with only the task window (in this case showing the extraction task) and as soon as the task ends, Keka will close.

When you say Keka open the compression window at every launch, you mean the advanced window?

![screen shot 2017-06-05 at 10 24 36](https://cloud.githubusercontent.com/assets/480163/26776370/285ef752-49d9-11e7-9567-ca50a2c0f658.png)

If this is the case, just note that you don't need to open Keka before doing an operation. You can drag and drop files to the icon in the dock, you can just double click for extracting files or use the [context menu extension](https://github.com/aonez/Keka/releases/download/v1.0.8/Legacy-ContextMenu-1.3.7z). Doing all of this do no require to open first Keka.

If the problem is that even with all the windows shut, the app keeps open in the dock, let me know.

**Comment by kristofferR** [comment]
Here's how it works for me:
https://www.youtube.com/watch?v=Kmnb3lPO-08

**Comment by aonez** [comment]
Can you zip (to attach here later If necessary) and delete this file:

`/Users/YOURUSERNAME/Library/Preferences/com.aone.keka.plist`

Be sure to close Keka first. Then try again. If that solves the issue, please attach the zipped file so I can see what was the problem.

**Comment by kristofferR** [comment]
[com.aone.keka.plist.zip](https://github.com/aonez/Keka/files/1051951/com.aone.keka.plist.zip)

No difference after deleting the preference plist (or its caches).

**Comment by aonez** [comment]
Just noticed you're using the 1.1 beta, is that right? Can you reproduce this with the stable version? (if needed, delete the plist before testing the stable one).

**Comment by kristofferR** [comment]
Yeah, no difference with 1.0.8.

**Comment by aonez** [comment]
So strange. Can you take a look at the Console.app (you can filer with "Keka") and let me know what it says?

You can enable more message output like this:
`defaults write com.aone.keka DevLog -bool TRUE`

Just disable the extra message output with FALSE:
`defaults write com.aone.keka DevLog -bool TRUE`

**Comment by kristofferR** [comment]
```standard	11:58:41.604083 +0200	Keka	Debug mode enabled
standard	11:58:41.827686 +0200	Keka	Arguments: (
    x,
    "/Users/kristoffer/Downloads/Browsers/Oxidizer_0_8_1.zip",
    "-o/Users/kristoffer/Downloads/Browsers/.Keka-56D70671-68BA-44A0-BA0E-5114118B3410",
    "-aou",
    "-xr!__MACOSX"
)
standard	11:58:41.827756 +0200	Keka	Binary used: keka7z
standard	11:58:41.847431 +0200	Keka	Extraction output: 

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=utf8,Utf16=on,HugeFiles=on,64 bits,4 CPUs x64)
Modified by aONe for Keka

Scanning the drive for archives:
  0M Scan /Users/kristoffer/Downloads/Browsers/
standard	11:58:41.849402 +0200	Keka	Extraction output: 
\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H                                               \^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H1 file, 5983860 bytes (5844 KiB)

Extracting archive: /Users/kristoffer/Downloads/Browsers/Oxidizer_0_8_1.zip
--
Path = /Users/kristoffer/Downloads/Browsers/Oxidizer_0_8_1.zip
Type = zip
Physical Size = 5983860

  0%
standard	11:58:42.044603 +0200	Keka	Extraction output: 
\^H\^H\^H\^H    \^H\^H\^H\^H 95% 274 - Oxidizer/test_scripts/_tutorial1.flam3
standard	11:58:42.058182 +0200	Keka	Extraction output: 
\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H                                                 \^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H\^H
standard	11:58:42.058276 +0200	Keka	Extraction output:
standard	11:58:42.334687 +0200	Keka	done: 1
standard	11:58:42.367267 +0200	Keka	growl++ = 0, blocking: 0, waiting: 0```

**Comment by aonez** [comment]
No prior messages to that? When Keka is opened, some messages are printed, those are the ones I'm interested in. Thanks for your time @kristofferR

**Comment by kristofferR** [comment]
That's the first messaged from Kaka in my Console. I've tried again after a reboot and with different files,and the log still looks like that.

**Comment by aonez** [comment]
One of the first lines should be: `Keka[73402:847391] Debug mode enabled`

Are you using a custom macOS?

Please do as follows:

1. Close Keka, if opened.
2. Delete **com.aone.keka.plist** as stated [before](https://github.com/aonez/Keka/issues/38#issuecomment-306149895)
3. Open Keka latest stable version, which right now is 1.0.8.

Let me know if the first time you open Keka you see the preferences and the advanced window.

Also I noticed you have a lot of in-dock apps. Maybe some of them are for handling windows? Just trying to remove things off the equation. Maybe creating a new Mac user, just for test purposes, will help.

**Comment by plam4u** [comment]
Same issue happens for me and I'm thinking to stop using Keka :(
How can I configure Keka to auto-close when a file is decompressed?
I've tried to delete the plist file. Also, today it updated to 1.0.10.

**Comment by aonez** [comment]
@plam4u what version of macOS are you using? Happens the same as in https://github.com/aonez/Keka/issues/38#issuecomment-306136289 video?

**Comment by plam4u** [comment]
@aonez, yes, the same thing happens to me. 
I'm using version MacOS Sierra 10.12.6 Beta (16G12b).

**Comment by Spartinus** [comment]
I'm having this issue as well on High Sierra

**Comment by aonez** [comment]
Could you try [this beta](http://www.kekaosx.com/release/test/Keka-1.1.0-beta.4-advancedwindow.7z), it will create a "Keka-log.log" file in your desktop. Please try to replicate this issue and send me that log file (zip compressed, so GitHub accepts it).

**Comment by plam4u** [comment]
I've uninstalled Keka. Sorry, I've tried to fix this issue for several months now. 
Unsubscribing from this issue.

**Comment by aonez** [comment]
Sad to hear that, @plam4u. Unable to reproduce it yet.

**Comment by kristofferR** [comment]
The beta solved this issue for me. I also like the new icons 👍

**Comment by aonez** [comment]
Glad to hear that @kristofferR! I'll update you all with the next beta to see if it does not reappear, since I'm yet unable to reproduce it and can't find where the problem is.

**Comment by hashb** [comment]
Disabling [afloat](https://github.com/w0lfschild/mySIMBL) fixes this issue

I was able to reproduce this bug with both the stable and beta releases

**Comment by aonez** [comment]
Thanks @hashb for the tip! Gonna try this to replicate.
@kristofferR & @Spartinus do you also use/used this `afloat` app?

**Comment by aonez** [comment]
> Disabling afloat fixes this issue

@hashb I get that you'll an option for Keka to be always on top?

**Comment by hashb** [comment]
No, I haven't activated 'always on top' for keka, I am not sure why this is happening.

**Comment by kristofferR** [comment]
Ah, I used afloat too. I never set keka to be always on top neither.

**Comment by jordy-el** [comment]
> Disabling afloat fixes this issue

I didn't want to disable afloat, so I snooped around for a better option and discovered that you can tell mySIMBL to just ignore specific apps instead. You won't be able to use any other mySIMBL plugins in Keka, but to me that's a better option than disabling it entirely.

1. Open mySIMBL
2. In the sidebar, open the SIMBL menu
3. Under SIMBL application blacklist, find and tick the checkbox for Keka

Keka should then close as it is supposed to.

**Comment by aonez** [comment]
Fixed in 1.0.14
